docs/website/contents/for-developers/HardForkCombinatorRollbackMonotonicityProposal.md

```diff
@@ -0,0 +1,190 @@
+## Though Experiment: Predictions within a Simplified Node
+
+Consider a simplified node.
+
+- Only node-to-node protocols: no queries, etc.
+- Pure Praos: no hard fork combinator, no Transitional Praos, etc.
+
+### Question 1
+
+Which aspects of the protocol and ledger's future behavior must this node be able to predict from a given ledger state and why?
+
+- Assume it's well-connected and running in a healthy net.
+- Assume k > 0.
+- Let D be the node's currently selected chain.
+- Let C be the current best valid chain in the net.
+- Let L be the ledger state resulting from the youngest block that is on both C and D.
+- Let R count the number of blocks after L on D.
+  Observe that there must be more than R blocks after L on C, since C is better than D.
+  (This isn't necessarily true, due to tiebreakers.
+  However, even with tiebreakers, this remains true in the worst case, which is what ultimately matters.)
+- Common Prefix ensures R <= k.
+- L must contain enough information for the node to correctly judge the validity of the k+1th block after L on C.
+- Otherwise the Header-Body Split would prevent the node from adopting C, even though doing so requires at most rolling back k blocks.
```

Comment:
Perhaps the "Otherwise" bullet can be merged with the previous one, or indented one level.